deku-p/src/core/protocol/protocol.ml

```diff
@@ -73,7 +73,21 @@ let apply_operation ~current_level protocol operation :
             | Ok ledger -> (ledger, Some receipt, vm_state, None)
             | Error error -> (ledger, Some receipt, vm_state, Some error))
         | Operation_vm_transaction { sender; operation } -> (
-            let receipt = Receipt.Vm_transaction_receipt { operation = hash } in
+            let open Deku_ledger in
+            let contract_address =
+              match operation.operation with
+              | View { address; _ } -> address
+              | Originate _ ->
+                  Contract_address.of_user_operation_hash
+                  @@ Operation_hash.to_blake2b hash
+              | Call { address; _ } ->
+                  Deku_ledger.Address.to_contract_address address
+                  |> Option.map fst |> Option.get
```

Comment:
This will throw an exception, and the user will not get any notification. This could be improved by giving the user an error receipt instead.